components/script/body.rs

```diff
@@ -59,15 +59,14 @@ pub fn consume_body<T: BodyTrait + Reflectable>(object: &T, body_type: BodyType)
                                                       body_type,
                                                       object.get_mime_type());
     match pkg_data_results {
-        Err(e) => promise.maybe_reject_error(promise.global().r().get_cx(), e),
+        Err(e) => promise.reject_error(promise.global().r().get_cx(), e),
         Ok(r) => {
-            let d = match r {
-                FetchedData::Text(s) => USVString(s),
-                FetchedData::Json(j) => j,
-                FetchedData::BlobData(b) => b,
-                FetchedData::FormData(f) => f,
+            match r {
+                FetchedData::Text(s) => promise.resolve_native(promise.global().r().get_cx(), &USVString(s)),
+                FetchedData::Json(j) => promise.resolve_native(promise.global().r().get_cx(), &j),
+                FetchedData::BlobData(b) => promise.resolve_native(promise.global().r().get_cx(), &b),
+                FetchedData::FormData(f) => promise.resolve_native(promise.global().r().get_cx(), &f),
```

Comment:
It may be possible to write this as:
```rust
let val: &ToJSConvertible = match r {
    FetchedData::Text(ref s) => s,
    FetchedData::Json(ref j) => j,
    FetchedData::BlobData(ref b) => b,
    FetchedData::FormData(ref f) => f,
};
promise.resolve_native(promise.global().r().get_cx(), v);
```